dm/ctl/master/query_status.go

```diff
@@ -60,5 +75,75 @@ func queryStatusFunc(cmd *cobra.Command, _ []string) {
 		return
 	}
 
-	common.PrettyPrintResponse(resp)
+	if resp.Result && taskName == "" && len(workers) == 0 {
+		result := wrapTaskResult(resp)
+		common.PrettyPrintInterface(result)
+	} else {
+		common.PrettyPrintResponse(resp)
+	}
+}
+
+// errorOccurred checks ProcessResult and return true if some error occurred
+func errorOccurred(result *pb.ProcessResult) bool {
+	return result != nil && len(result.Errors) > 0
+}
+
+// getRelayStage returns current relay stage (including stageError)
+func getRelayStage(relayStatus *pb.RelayStatus) string {
+	if errorOccurred(relayStatus.Result) {
+		return stageError
+	}
+	return relayStatus.Stage.String()
+}
+
+// wrapTaskResult picks task info and generate tasks' status and relative workers
+func wrapTaskResult(resp *pb.QueryStatusListResponse) *taskResult {
+	taskStatusMap := make(map[string]string, len(resp.Workers))
+	taskCorrespondingWorkers := make(map[string][]string, len(resp.Workers))
```

Comment:
A little weird here, both `taskStatusMap` and `taskCorrespondingWorkers` are mapping from taskName to something, the capacity of map doesn't equal to `len(resp.Workers)`, use zero is ok

Reply:
Revised in https://github.com/pingcap/dm/pull/334/commits/6856319104013b88d10b25a3b53a5770ca781b7a.